src/main/java/org/opensearch/knn/plugin/transport/KNNProfileResponse.java

```
@@ -0,0 +1,166 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.transport;
+
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.opensearch.action.support.broadcast.BroadcastResponse;
+import org.opensearch.core.action.support.DefaultShardOperationFailedException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.profiler.SegmentProfilerState;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class KNNProfileResponse extends BroadcastResponse implements ToXContentObject {
+
+    List<KNNIndexShardProfileResult> shardProfileResults;
+
+    public KNNProfileResponse() {}
+
+    public KNNProfileResponse(StreamInput in) throws IOException {
+        super(in);
+        int size = in.readInt();
+        shardProfileResults = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            shardProfileResults.add(new KNNIndexShardProfileResult(in));
+        }
+    }
+
+    public KNNProfileResponse(
+        List<KNNIndexShardProfileResult> shardProfileResults,
+        int totalShards,
+        int successfulShards,
+        int failedShards,
+        List<DefaultShardOperationFailedException> shardFailures
+    ) {
+        super(totalShards, successfulShards, failedShards, shardFailures);
+
+        this.shardProfileResults = shardProfileResults;
+    }
+
+    @Override
+    public void writeTo(StreamOutput streamOutput) throws IOException {
+        super.writeTo(streamOutput);
```

Let's see if we use this override for anything. If we don't we can throw an exception here to make sure we don't stray off the intended path